redash_toolbelt/examples/find-table-names.py

```diff
@@ -0,0 +1,204 @@
+    """Builds a summary showing table names and count of queries that reference them."""
+
+    summary = {
+        table_name: sum(
+            [1 for tables in tables_by_qry.values() if table_name in tables]
+        )
+        for table_name in itertools.chain(*tables_by_qry.values())
+    }
+
+    align = max([len(table_name) for table_name in summary.keys()])
+
+    print("\n")
+    print(f"{'table':>{align}} | {'number of queries':>17}")
+    print("-" * align + " | " + "-" * 17)
+
+    for t, num in sorted(summary.items(), key=lambda item: item[1], reverse=True):
+        print(f"{t:>{align}} | {num:>17}")
+
+    print("\n")
+
+
+def print_details(tables_by_qry):
+    """Prints out (query_id, tablename) tuples"""
+
+    details = [
+        [(query, table) for table in tables] for query, tables in tables_by_qry.items()
+    ]
+
+    for row in itertools.chain(*details):
+        print(",".join([str(i) for i in row]))
+
+
+@click.command()
+@click.argument("url",)
+@click.argument("key",)
+@click.argument("data_source_id")
+@click.option("--detail", is_flag=True, help="Prints out all table/query pairs?")
+def main(url, key, data_source_id, detail):
+    """Find table names referenced in queries against DATA_SOURCE_ID"""
+
+    data = find_table_names(url, key, data_source_id)
+
+    if detail:
+        print_details(data)
+    else:
+        print_summary(data)
+
+
+if __name__ == "__main__":
+    main()
+
+
+def test_1():
+
+    sql = """
+    SELECT field FROM table0 LEFT JOIN table1 ON table0.field = table1.field
+    """
+
+    tables = extract_table_names(sql)
+    expected = ["table0", "table1"]
+
+    assert len(tables) == len(expected) and all([i in expected for i in tables])
+
+
+def test_2():
+
+    sql = """
+    SELECT field FROM table0 as a LEFT JOIN table1 as b ON a.field = b.field
+    """
+
+    tables = extract_table_names(sql)
+    expected = ["table0", "table1"]
+
+    assert len(tables) == len(expected) and all([i in expected for i in tables])
+
+
+def test_3():
+
+    sql = """
+    SELECT field FROM table0 a LEFT JOIN table1 b ON a.field = b.field
+    """
+
+    tables = extract_table_names(sql)
+    expected = ["table0", "table1"]
+
+    assert len(tables) == len(expected) and all([i in expected for i in tables])
+
+
+def test_4():
+
+    sql = """
+    SELECT field FROM schema.table0 a LEFT JOIN schema.table1 b ON a.field = b.field
+    """
+
+    tables = extract_table_names(sql)
+    expected = ["schema.table0", "schema.table1"]
+
+    assert len(tables) == len(expected) and all([i in expected for i in tables])
+
+
+def test_5():
+
+    sql = """
+    SELECT field
+    FROM
+        table0
+    LEFT JOIN
+        table1
+    """
+
+    tables = extract_table_names(sql)
+    expected = ["table0", "table1"]
+
+    assert len(tables) == len(expected) and all([i in expected for i in tables])
+
+
+def test_6():
+
+    sql = """
+    SELECT field FROM table1,table2, table3 ,table4 , table5
```

Comment:
I think you need one more test:

```sql
FROM table1 AS t1, table2 AS t2, table3 t3, table4 t4
```

Reply:
Done. I added `test_8` which matches your statement.

I also added `test_9` which covers the combination of `test_8` mixed any traditional JOINS that follow a comma-delimited JOIN statement.